indico/modules/events/papers/client/js/components/RevisionJudgment.jsx

```diff
@@ -0,0 +1,125 @@
+
+import {resetPaperJudgment} from '../actions';
+import {getPaperDetails, getPaperPermissions, isPaperStateResetInProgress} from '../selectors';
+import UserAvatar from './UserAvatar';
+
+export default function RevisionJudgment({revision}) {
+  const {state, judge, isLastRevision, judgmentComment, judgmentDt} = revision;
+  const {event, contribution} = useSelector(getPaperDetails);
+  const [confirmOpen, setConfirmOpen] = useState(false);
+  const {canJudge} = useSelector(getPaperPermissions);
+  const isResetInProgress = useSelector(isPaperStateResetInProgress);
+  const dispatch = useDispatch();
+
+  if (state === 'submitted') {
+    return null;
+  }
+
+  return (
+    <div className="i-timeline-item">
+      <UserAvatar user={judge} />
+      <div className="i-timeline-item-box header-indicator-left">
+        <div className="i-box-header flexrow">
+          <div className="f-self-stretch">
+            {state === 'accepted' && (
+              <Translate>
+                <Param name="judgeName" value={judge.fullName} wrapper={<strong />} /> accepted this
+                paper.
+              </Translate>
+            )}
+            {state === 'rejected' && (
+              <Translate>
+                <Param name="judgeName" value={judge.fullName} wrapper={<strong />} /> rejected this
+                paper.
+              </Translate>
+            )}
+            {state === 'to_be_corrected' && (
+              <Translate>
+                <Param name="judgeName" value={judge.fullName} wrapper={<strong />} /> asked for
+                changes.
+              </Translate>
+            )}{' '}
+            <time dateTime={serializeDate(judgmentDt, moment.HTML5_FMT.DATETIME_LOCAL)}>
+              {serializeDate(judgmentDt, 'LL')}
+            </time>
+          </div>
+          {canJudge && isLastRevision && (
+            <>
+              <div className="hide-if-locked">
+                <a
+                  href="#"
+                  className="i-link icon-remove"
+                  title={Translate.string('Reset judgment')}
+                  onClick={() => setConfirmOpen(true)}
+                />
+              </div>
+              <Confirm
+                header={
+                  <Header icon="warning sign" content={Translate.string('Confirm the operation')} />
+                }
+                open={confirmOpen}
+                content={Translate.string(
+                  'Do you really want to reset the judgment? This operation is irreversible.'
+                )}
+                onCancel={() => setConfirmOpen(false)}
+                onConfirm={() => {
+                  dispatch(resetPaperJudgment(event.id, contribution.id)).then(({error}) => {
+                    // only hide the modal in case there was an error. This will prevent calling `setState` on
+                    // an already unmounted component.
+                    if (error) {
+                      setConfirmOpen(false);
```

Comment:
actually does it even make sense to close the confirm in case of an error? I'd expect it to be open so I can try again if it failed..